### PR TITLE
raftstore: remove is_in_flashback field in peer fsm

### LINKED ISSUE #13868: after flashback, admin check table report " region 20009 is in flashback progress"
### What version of TiKV are you using?
[root@localhost test_kv]# ./tikv-server -V
TiKV 
Release Version:   6.5.0-alpha
Edition:           Community
Git Commit Hash:   46f9a576e785c5006e9390ebd57623f5d3a6a5cc
Git Commit Branch: flashback_unify_reader
UTC Build Time:    2022-12-01 03:28:15
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           release
[root@localhost test_kv]# 

### Steps to reproduce
1. tidb cluster with tiflash and pitr task and ticdc task
2. prepare 300G data
3. run update with 2k qps for 1 hour
4. flashback to 1 hour ago

### What did you expect?
flashback successfully
data consistent in upstream and downstream

### What did happened?
```code 
MySQL [flashback]> admin check table user_data4 ;
ERROR 1105 (HY000): region 31259 is in flashback progress

MySQL [(none)]> flashback cluster to timestamp '2022-12-01 17:50:15';
Query OK, 0 rows affected (29 min 42.77 sec)
MySQL [flashback]> admin check table user_data1 ;
ERROR 1105 (HY000): region 28192 is in flashback progress
MySQL [flashback]> admin check table user_data2 ;
Query OK, 0 rows affected (3.32 sec)

MySQL [flashback]> admin check table user_data3 ;
Query OK, 0 rows affected (4.19 sec)

MySQL [flashback]> admin check table user_data4 ;
ERROR 1105 (HY000): region 31259 is in flashback progress
MySQL [flashback]> admin check table user_data5 ;
ERROR 1105 (HY000): region 23474 is in flashback progress
MySQL [flashback]> admin check table user_data6 ;
Query OK, 0 rows affected (2.74 sec)

MySQL [flashback]> admin check table user_data7 ;
Query OK, 0 rows affected (3.55 sec)

MySQL [flashback]> admin check table user_data8 ;
Query OK, 0 rows affected (4.77 sec)

MySQL [flashback]> admin check table user_data9 ;
ERROR 1105 (HY000): region 20009 is in flashback progress
MySQL [flashback]> admin check table user_data10;
Query OK, 0 rows affected (4.91 sec)

MySQL [flashback]>
MySQL [flashback]> admin show ddl jobs;
+--------+-----------+-------------+-------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
| JOB_ID | DB_NAME   | TABLE_NAME  | JOB_TYPE          | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE  |
+--------+-----------+-------------+-------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
|    182 |           |             | flashback cluster | public       |         0 |        0 |         0 | 2022-12-01 20:07:08 | 2022-12-01 20:07:09 | 2022-12-01 20:36:51 | synced |
```

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: Close #13868.

What's Changed:

```commit-message
- Remove `is_in_flashback` field and use the region meta as the only source of truth in `PeerFSM`.
- Add a corresponding test case.
- Some minor refinement to the code and tests.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -4738,6 +4738,9 @@ where
             "region" => ?region,
         );
 
+        // Update the flashback flag according to the snapshot region meta.
+        self.fsm.peer.is_in_flashback = region.is_in_flashback;
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:4742]: Remove `is_in_flashback` field and use `region.is_in_flashback` as only source of truth. `set_region` would update it after applying snapshot. And `on_set_flashback_state` should update region as well.

Comment (JmPotato) [reply components/raftstore/src/store/fsm/peer.rs]: Nice idea. Already addressed. PTAL.

File: src/storage/txn/commands/flashback_to_version_read_phase.rs [hunk 2]
```diff
@@ -169,6 +171,11 @@ impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
                         commit_ts: self.commit_ts,
                     }));
                 }
+                // Filter out the SST that does not have a newer version than `self.version` in
+                // `CF_WRITE`, i.e, whose latest `commit_ts` <= `self.version` in the later
+                // scan. By doing this, we can only flashback those keys that have version
+                // changed since `self.version` as much as possible.
+                reader.set_hint_min_ts(Some(Bound::Excluded(self.version)));
```
Comment (JmPotato) [inline src/storage/txn/commands/flashback_to_version_read_phase.rs:178]: This should be set globally inside the `process_read`.

**Review by fengou1 — APPROVED** [review]
EBS verification on the PR:  the pr test has passed the checksum validation during the backup and restore, also showing the great performance.

**Review by HuSharp — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]
LGTM

**Comment by JmPotato** [comment]
@Connor1996 PTAL again, thx.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**